### PR TITLE
Remove the t-shirt form

### LINKED ISSUE #1771: Remove the t-shirt form
There's currently a form [1][2] on AMO that is used to allow contributors to request a t-shirt when their add-on or theme reaches certain criteria. 

Hopefully we will soon be specifying form-src in our CSP policy. Ideally it would be nice to be able to lock down form posts to just the main origin of a.m.o rather than additionally allow posts to google (the t-shirt form sends the data to docs.google.com).

Is this used - and can we consider removing this feature and doing this some other way?

[1] https://github.com/mozilla/addons-server/blob/e66197c915d5b260645fd73fec04ab22b5db7a22/src/olympia/users/views.py#L263
[2] https://github.com/mozilla/addons-server/blob/76f1143fc1409dbfed4f94f5bba2b6961c0efefd/src/olympia/users/templates/users/t-shirt.html#L49

### PR DESCRIPTION
**Pull request by lavish205**
Fixes #1771

### CODE REVIEW

**Comment by wagnerand** [comment]
Pro tip: If you say "Fixes #1771" instead of just mentioning the issue, it will close the issue automatically when the PR gets merged.

**Comment by wagnerand** [comment]
There is a a bit to remove than that.
You probably want to `git diff ecd8b21^1 64c6df8`

**Comment by wagnerand** [comment]
I think you missed the model:

src/olympia/users/models.py
201:    t_shirt_requested = models.DateTimeField(blank=True, null=True,

File: src/olympia/migrations/879-remove-t-shirt-order-field.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+ALTER table `users` 
+    DROP column `t_shirt_requested`;
```
Comment (eviljeff) [inline src/olympia/migrations/879-remove-t-shirt-order-field.sql]: can you add `DELETE FROM waffle_switch WHERE name = 't-shirt-orders';` to remove the waffle switch too?

**Comment by lavish205** [comment]
Fixed

File: src/olympia/migrations/879-remove-t-shirt-order-field.sql (hunk 1 above)
Comment (lavish205) [inline src/olympia/migrations/879-remove-t-shirt-order-field.sql]: Done

**Comment by eviljeff** [comment]
lgtm - when the tests pass we can merge
r+

**Pull request merged**